### PR TITLE
feat: swap execution hook for multi-hop trades

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Implements swap execution hook for multi-hop trades.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

progresses #5653

## Risk

Very low risk to production facing trades, but worth sanity checking in case something is broken.

## Testing

Quick sanity check that old UI trades are still ok.

## Screenshots (if applicable)

Tested single hop trade with approval on new UI using new hooks:
image
image
image
image
image
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5774** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5774?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5781** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5781?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5782** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5782?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5781?utm_source=stack-comment).

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  }, [])
+
+  const executeTrade = useCallback(() => {
+    if (!wallet) throw Error('missing wallet')
+    if (!accountMetadata) throw Error('missing accountMetadata')
+    if (!tradeQuote) throw Error('missing tradeQuote')
+    if (!swapperName) throw Error('missing swapperName')
+    if (!sellAssetAccountId) throw Error('missing sellAssetAccountId')
+
+    return new Promise<void>(async (resolve, reject) => {
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx:65]: 🧠

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            tradeQuote,
+            stepIndex: hopIndex,
+            slippageTolerancePercentageDecimal,
+            from,
+            signAndBroadcastTransaction: async (transactionRequest: StdSignDoc) => {
+              const txToSign: ThorchainSignTx = {
+                addressNList: toAddressNList(bip44Params),
+                tx: {
+                  fee: {
+                    amount: [...transactionRequest.fee.amount],
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx:206]: Remove spread?

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                    gas: transactionRequest.fee.gas,
+                  },
+                  memo: transactionRequest.memo,
+                  msg: [...transactionRequest.msgs],
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx:210]: Remove spread?

**Review by gomesalexandre — COMMENTED** [review]
Conceptually looks sane to me, will stamp after a sanity runtime test

**Review by gomesalexandre — APPROVED** [review]
Sanity tested locally, confimed swap execution state is still looking gucci with and without allowance granted:

### Allowance granted

Screenshot 2023-12-06 at 09 09 14
Screenshot 2023-12-06 at 09 09 14

### Approval required

Screenshot 2023-12-06 at 09 12 20
Screenshot 2023-12-06 at 09 12 27
Screenshot 2023-12-06 at 09 12 56
Screenshot 2023-12-06 at 09 13 21

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx (hunk 2 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx:206]: alas no , she's readonly
```
StdFee.amount: readonly Coin[]
```

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx (hunk 3 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx:210]: same here, readonly `StdSignDoc.msgs: readonly Msg[]`

**Pull request merged**